ansible_runner/utils/streaming.py

```diff
@@ -0,0 +1,59 @@
+import tempfile
+import zipfile
+import os
+import json
+import sys
+
+from .base64io import Base64IO
+from pathlib import Path
+
+def stream_dir(source_directory, stream):
+    with tempfile.NamedTemporaryFile() as tmp:
+        with zipfile.ZipFile(tmp.name, 'w', compression=zipfile.ZIP_DEFLATED, allowZip64=True) as archive:
+            if source_directory:
+                for dirpath, dirs, files in os.walk(source_directory):
+                    relpath = os.path.relpath(dirpath, source_directory)
+                    if relpath == ".":
+                        relpath = ""
+                    for fname in files:
+                        archive.write(os.path.join(dirpath, fname), arcname=os.path.join(relpath, fname))
+            archive.close()
+
+        zip_size = Path(tmp.name).stat().st_size
```

Comment:
wait, did you lose the calculation of tarball -> base64 size?

Reply:
Yep! It turns out that `base64io` is smart enough that when you say you need to read `n` number of bytes, it gives you that many bytes of _unencoded_ data. I thought that was pretty neat.